pkg/datapath/loader/base.go

```diff
@@ -231,17 +230,14 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 			}
 		}
 
-		if option.Config.DatapathMode == datapathOption.DatapathModeIpvlan {
-			mode = "ipvlan"
+		if option.Config.Tunnel != option.TunnelDisabled {
```

Comment:
Hmm, unrelated to your PR, but I'm wondering whether we could move it completely away outside the `if len(option.Config.Devices) != 0` to the line 223, get rid of the tunnel setting at 234, and also move away the flannel setting (the line 246) outside the if-block.

Reply:
Yep, it looks like the `args[initArgMode]` assignments could be simplified a lot (also the `"routed"` assignment on line 253).
Since the tests started and this fix is required for two different `priority/release-blocker` PRs, maybe we can do that as a followup? I'll add a TODO to my list to not forget.

Reply:
Sure, let's do it as a follow-up.